tool.c

```diff
@@ -126,6 +127,48 @@ clean(int argc, char *argv[])
 	return ret;
 }
 
+/* Depth-first search.  */
+static void
+targetcommands(struct node *n)
+{
+	struct edge *e = n->gen;
+	struct string *command;
+	size_t i;
+	if (!e || (e->flags & FLAG_DIRTY_OUT))
+		return;
+	e->flags |= FLAG_DIRTY_OUT;
```

Comment:
Not that it matters, but I'd use `FLAG_WORK` here, since it seems to be closest to what we want here.